crates/bevy_ui/src/render/ui_texture_slice_pipeline.rs

```diff
@@ -715,10 +727,10 @@ fn compute_texture_slices(
                 1. - (border.bottom / target_size.y) * min_coeff,
             ];
 
-            let isx = image_size.x * (1. - slices[0] - slices[2]);
-            let isy = image_size.y * (1. - slices[1] - slices[3]);
-            let tsx = target_size.x * (1. - border[0] - border[2]);
-            let tsy = target_size.y * (1. - border[1] - border[3]);
+            let isx = image_size.x * (slices[2] - slices[0]);
+            let isy = image_size.y * (slices[2] - slices[1]);
```

Comment:
Don't know much about nine-patch, but the code before used `3` instead of `2` on this line.
Can't comment on the mathematical correctness of this change in general. It looks like it's doing something else? Could you comment on that?

Reply:
In the old texture slicing implementation (which used a separate sprite for each slice) insets were used to define the slices, but I found it more natural to use normalized coordinates, offset from the the top-left corner.

So previously nine slicing a 100x100 image with 25x25 border corner pieces you had
```
border = [
    0.25 // left inset,
    0.25 // top inset,
    0.25 // right inset,
    0.25 // bottom inset,
]
```
and then to calculate the normalized width of the horizontal side and center sections it would do:
`1 - border[0] - border[2] = 1 - 0.25 - 0.25 = 0.5`

whereas now it's
```
border = [
    0.25 // top-left x,
    0.25 // top-left y,
    0.75 // bottom-right x,
    0.75 // bottom-right y,
]
```
and to get the normalized width it's
`border[2] - border[0] = 0.75 - 0.25 = 0.5`

Reply:
Does that affect how users define the slices? Will it break current usage ?

Reply:
No the user API should behave the same as before, the coordinates change is just for the internal calculations.